### ISSUE TITLE
GUI appears on wrong monitor (wayland)

### ISSUE THREAD

**Issue by ParadoxLeon** [flameshot-org/flameshot#3686]
### Flameshot Version

Flameshot v12.1.0

### Installation Type

User repository (AUR)

### Operating System type and version

EndeavourOS

### Description

When I press print to take a screenshot flameshot opens on my second monitor every.

### System Information

Operating System: EndeavourOS 
KDE Plasma Version: 6.1.3
KDE Frameworks Version: 6.4.0
Qt Version: 6.7.2
Kernel Version: 6.10.2-arch1-1 (64-bit)
Graphics Platform: Wayland
Processors: 12 × AMD Ryzen 5 5600X 6-Core Processor
Memory: 31,3 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 2070/PCIe/SSE2
Manufacturer: Micro-Star International Co., Ltd.
Product Name: MS-7C37
System Version: 1.0

**Comment by jclsn** [comment]
I am experiencing the same issue. Makes the application unusable on Wayland.

**Comment by Goddard** [comment]
Same issue here.

Operating System: Manjaro Linux 
KDE Plasma Version: 6.0.5
KDE Frameworks Version: 6.4.0
Qt Version: 6.7.2
Kernel Version: 6.8.12-3-MANJARO (64-bit)
Graphics Platform: Wayland
Processors: 32 × AMD Ryzen 9 7950X3D 16-Core Processor
Memory: 30.5 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 4090/PCIe/SSE2
Manufacturer: Gigabyte Technology Co., Ltd.
Product Name: X670E AORUS PRO X
System Version: -CF

**Comment by natewu** [comment]
Same issue here, keeps opening on my second monitor then I clicked it once and somehow now it only opens on my main monitor and wont focus on the correct monitor. I remember on X both monitors are focused so I could screenshot on both monitors but it appears that is missing too.

Operating System: Arch Linux 
KDE Plasma Version: 6.1.4
KDE Frameworks Version: 6.4.0
Qt Version: 6.7.2
Kernel Version: 6.10.4-arch2-1 (64-bit)
Graphics Platform: Wayland
Processors: 12 × AMD Ryzen 5 3600X 6-Core Processor
Graphics Processor: AMD Radeon RX 7600

**Comment by DennisDyallo** [comment]
Happens on Windows 11 to, even though I have correctly set my Main Display.

**Comment by ParadoxLeon** [comment]
### Current workaround (linux)

clone the repo `git clone https://github.com/flameshot-org/flameshot.git` and cd into it.
change to fix_wayland_multiscreen with `git checkout fix_wayland_multiscreen`
install build [dependencies ](https://github.com/flameshot-org/flameshot/tree/fix_wayland_multiscreen?tab=readme-ov-file#compilation).
and follow the [build](https://github.com/flameshot-org/flameshot/tree/fix_wayland_multiscreen?tab=readme-ov-file#build) process from there on.

Done. When opening with a shortcut I get some short flickers sometimes but other than that it works :)

**Comment by Goddard** [comment]
Doesn't work great for me.

What happens for me is it always launches on two of my three monitors.  It is always the same two.  Seems to just ignore the other.

It seems like it isn't even the correct two display images either.  

So it shows the first monitor in the second place and the second monitor in the third place monitor.  So seems to still have some issue.

**Comment by ParadoxLeon** [comment]
weird it say's the same for me but it works on my end...

**Comment by mmahmoudian** [comment]
The [v14.0.0](https://github.com/flameshot-org/flameshot/releases/tag/v14.0.0) is now released which has solved this issue.

**Comment by GregTheMadMonk** [comment]
~~Tried it and it indeed captures the active monitor correctly.~~

~~But does that mean that across-monitor screenshots are impossible now?~~

I confused this issue for another one and was talking about X11

**Comment by mmahmoudian** [comment]
@GregTheMadMonk 
> But does that mean that across-monitor screenshots are impossible now?

Well, yes and no 😅  You still can take a full screenshot of all your monitors using `full` subcommand (e.g., `flameshot full --path /tmp/blah.png`) but you cannot annotate it. The annotation is now exclusively available in `flameshot gui` and `flameshot screen --edit`.

Perhaps in future we can find a way to do full screen annotation, but for the time being we haven't find a reliable way to do this in Wayland.

**Comment by GregTheMadMonk** [comment]
I'm sorry @mmahmoudian I've confused this issue with #3847 - I'm on X11 🤦 IDK why am I subscribed to this one - maybe I had issues with this during one of my annual Wayland test-drives? Eh, doesn't matter now.

But since we're here, is it correct that this feature is also missing on X11 in the new versions?

And sorry for the confusion :(

**Comment by mmahmoudian** [comment]
@GregTheMadMonk 
> s it correct that this feature is also missing on X11 in the new versions?

I believe so, but turn on the X11 legacy mode from config and test. Sorry, I don't remember it from the top of my head (I'm writing this on my phone).


> And sorry for the confusion :(

No worries at all. I'm also sometimes lost in the issues here. Every issue is based on a specific setup and it is sometimes hard to track where we are commenting.

**Comment by GregTheMadMonk** [comment]
> turn on the X11 legacy mode from config and test

Yeah, seems to be the case... guess I'll stick with an older version for now. Not that it's very useful very often tbh, but I'm just very used to it being this way

Hope you figure something out and bring it back! Thanks for this great tool!
